lib/rules/prefer-regex-literals.js

```diff
@@ -293,6 +296,51 @@ module.exports = {
             }
         }
 
+        /**
+         * Checks whether the given regex and flags are valid for the ecma version or not.
+         * @param {string} pattern The regex pattern to check.
+         * @param {string | undefined} flags The regex flags to check.
+         * @returns {boolean} True if the given regex pattern and flags are valid for the ecma version.
+         */
+        function isValidRegexForEcmaVersion(pattern, flags) {
+            const RegExpValidatorInstance = new RegExpValidator({ ecmaVersion: regexppEcmaVersion });
+
+            try {
+                RegExpValidatorInstance.validatePattern(pattern, 0, pattern.length, flags ? flags.includes("u") : false);
+                if (flags) {
+                    RegExpValidatorInstance.validateFlags(flags);
+                }
+                return true;
+            } catch {
+                return false;
+            }
+        }
+
+        /**
+         * Checks whether two given regex flags contain the same flags or not.
+         * @param {string} flagsA The regex flags.
+         * @param {string} flagsB The regex flags.
+         * @returns {boolean} True if two regex flags contain same flags.
+         */
+        function isFlagsEqual(flagsA, flagsB) {
+            return [...flagsA].sort().join("") === [...flagsB].sort().join("");
+        }
+
+
+        /**
+         * Merges two regex flags.
+         * @param {string} flagsA The regex flags.
+         * @param {string} flagsB The regex flags.
+         * @returns {string} The merged regex flags.
+         */
+        function mergeRegexFlags(flagsA, flagsB) {
+            return [
+                ...flagsA,
+                ...flagsB
+            ].filter((flag, index, self) => index === self.indexOf(flag))
```

Comment:
I think you are trying to de-duplicate here? It's not very clear. I'd suggest rewriting to use a `Set`:

```suggestion
            return [...new Set([
                ...flagsA,
                ...flagsB
            ])]
```